src/client/java/net/neoforged/neoforge/client/neo3d/GpuDeviceFeatures.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.neo3d;
+
+import java.lang.reflect.AccessFlag;
+import java.lang.reflect.Field;
+import org.jetbrains.annotations.ApiStatus;
+
+public class GpuDeviceFeatures implements Cloneable {
+    /**
+     * LogicOp is problematic on Qualcomm GPUs via OpenGL
+     * LogicOp is unavailable on MacOS via Vulkan
+     */
+    public boolean logicOp = false;
+
+    private static final Field[] fields = GpuDeviceFeatures.class.getFields();
+
+    @ApiStatus.Internal
+    public boolean hasAll(GpuDeviceFeatures features) {
+        try {
+            for (final var field : fields) {
```

Comment:
I don't like this usage of reflection.

Reply:
It's not my favorite either, but I felt it would be more reasonable than the `if` or `||`/`&&` hell this would otherwise grow to. Currently there is only one field, but it likely wont stay that way for long.